src/main/java/org/broadinstitute/consent/http/resources/DataSetResource.java

```diff
@@ -112,9 +113,14 @@ public Response createDataset(@Auth AuthUser user, @Context UriInfo info, String
         }
         User dacUser = userService.findUserByEmail(user.getGoogleUser().getEmail());
         Integer userId = dacUser.getDacUserId();
-        DataSet createdDataset = datasetService.createDataset(inputDataset, name, userId);
-        URI uri = info.getRequestUriBuilder().replacePath("api/dataset/{datasetId}").build(createdDataset.getDataSetId());
-        return Response.created(uri).entity(createdDataset).build();
+        DataSetDTO createdDataset = datasetService.createDataset(inputDataset, name, userId);
+        Consent createdConsent = datasetService.createConsentForDataset(createdDataset);
+        if (Objects.isNull(createdConsent)) {
+            throw new ClientErrorException("Consent was not able to be created for newly created dataset with id: " + createdDataset.getDataSetId(), Status.CONFLICT);
```

Comment:
Are we sure this is a conflict? Seems like this would be more of a server error.